sql/src/main/java/io/crate/execution/engine/collect/collectors/OrderedLuceneBatchIteratorFactory.java

```diff
@@ -52,10 +55,11 @@
 
     public static BatchIterator<Row> newInstance(List<OrderedDocCollector> orderedDocCollectors,
                                                  Comparator<Row> rowComparator,
+                                                 Supplier<RowAccounting> rowAccountingSupplier,
```

Comment:
Is there a reason to choose Supplier here instead of RowAccounting directly?

Reply:
just for ease of testing

Reply:
I don't really see how this makes testing easier?